kolibri/content/api.py

```diff
@@ -122,12 +122,16 @@ def filter_next_steps(self, queryset, value):
 
         completed_content_nodes = queryset.filter(content_id__in=completed_content_ids).order_by()
 
-        return queryset.exclude(pk__in=completed_content_nodes).filter(
+        # Filter to only show content that the user has not engaged in, so as not to be redundant with resume
+        engaged_content_nodes = queryset.filter(content_id__in=ContentSummaryLog.objects.filter(
+            user=value).values_list('content_id', flat=True))
+
+        return queryset.exclude(pk__in=engaged_content_nodes).filter(
```

Comment:
This would seem to imply that a nested `SELECT` will take place inside a nested `SELECT` for an exclusion - might be expensive.

Reply:
Yeah, can probably simplify this query slightly.